client/web/src/search/suggestion/SmartSearchPreview.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react'
+import { RouteComponentProps } from 'react-router'
+
+import { mdiArrowRight } from '@mdi/js'
+import classNames from 'classnames'
+
+import { of } from 'rxjs'
+import { tap } from 'rxjs/operators'
+
+import { SyntaxHighlightedSearchQuery, smartSearchIconSvgPath } from '@sourcegraph/branded'
+import { SearchPatternType } from '../../../../shared/src/graphql-operations'
+import { formatSearchParameters } from '@sourcegraph/common'
+import { LATEST_VERSION, aggregateStreamingSearch, ProposedQuery } from '../../../../shared/src/search/stream'
+import { SearchMode } from '@sourcegraph/shared/src/search'
+import { Link, Icon, H3, H2, Text, Button, createLinkUrl, useObservable } from '@sourcegraph/wildcard'
+
+import { useNavbarQueryState, setSearchMode } from '../../stores'
+import { submitSearch } from '../helpers'
+
+import styles from './QuerySuggestion.module.scss'
+
+interface SmartSearchPreviewProps extends Pick<RouteComponentProps, 'history'> {}
+
+export const SmartSearchPreview: React.FunctionComponent<React.PropsWithChildren<SmartSearchPreviewProps>> = ({
```

Comment:
```suggestion
export const SmartSearchPreview: React.FunctionComponent<{}> = ({
```

This component doesn't use the`children` prop, so using `PropsWithChildren` is not ideal. I think having the props be an empty object (`{}`) is OK.